Hymns/Tags/TagEntity.swift

```diff
@@ -34,8 +34,48 @@ class TagEntity: Object, Identifiable {
     static func createPrimaryKey(hymnIdentifier: HymnIdentifier, tag: String) -> String {
         return ("\(hymnIdentifier.hymnType):\(hymnIdentifier.hymnNumber):\(hymnIdentifier.queryParams ?? [String: String]()):\(tag)")
     }
+
+    override func isEqual(_ object: Any?) -> Bool {
+        return primaryKey == (object as? TagEntity)?.primaryKey
+    }
+
+    override var hash: Int {
+        return primaryKey.hash
+    }
 }
 
 enum TagColor: Int {
     case none, blue, green, yellow, red
 }
+
+extension TagColor {
+    var background: UIColor {
+        switch self {
+        case .none:
+            return UIColor.white
+        case .blue:
+            return UIColor(red: 2/255, green: 118/255, blue: 254/255, alpha: 0.2)
+        case .green:
+            return UIColor(red: 80/255, green: 227/255, blue: 194/255, alpha: 0.2)
+        case .yellow:
+            return UIColor(red: 255/255, green: 209/255, blue: 0/255, alpha: 0.2)
+        case .red:
+            return UIColor(red: 255/255, green: 0, blue: 31/255, alpha: 0.2)
+        }
+    }
+
+    var foreground: UIColor {
+        switch self {
+        case .none:
+            return UIColor.white
```

Comment:
use primary color

Reply:
I had to use label for the foreground since primary is not a UI Color

Reply:
Do these need to be UIColors? I think you can change them to just Color and use `Color(UIColor(...))`

Reply:
Nice! I don't know why I didn't think of that lol... Great idea. Done!